### ISSUE TITLE
Mastodon is sending repeated Delete activities for the same deleted account

### ISSUE THREAD

**Issue by aaronpk** [mastodon/mastodon#20406]
### Steps to reproduce the problem

1. delete an account on an instance
2. observe the Delete activities sent by the instance

### Expected behaviour

One Delete activity is sent

### Actual behaviour

Delete activities are sent indefinitely

### Detailed description

I looked at the logs of my (hand-rolled) ActvityPub inbox recently and noticed I've received an unusual amount of Delete messages from the same account.

Since April 2021, I've been receiving account delete messages for this user:

```json
{
    "@context": "https://www.w3.org/ns/activitystreams",
    "id": "https://mastodon.cc/users/alatiera#delete",
    "type": "Delete",
    "actor": "https://mastodon.cc/users/alatiera",
    "to": [
        "https://www.w3.org/ns/activitystreams#Public"
    ],
    "object": "https://mastodon.cc/users/alatiera",
    "signature": {
        "type": "RsaSignature2017",
        "creator": "https://mastodon.cc/users/alatiera#main-key",
        "created": "2021-04-06T14:48:38Z",
        "signatureValue": "zVcqR/mjfL1TidfIuNKcvHy+hxNjy9na5AAju0AxiGX2kUOqh9oD1jbvAWWFWL+eqq6/2hqI0pQXkF+edZxBtrii33gCqh1EPo02q0KkXtgUV60mkddIEMeVYDs+93xYC4j4sSP1CZYORC8hto6k/0Mz83Ruh5VgJXa6tgGeQYlRqn9pfwzwGkLLzsoGfKLAu3kQ3xKBCtBDn2xzgXIHYUjFGGwlLY47x2m0ghCqFBvYcU3/nn7irI+jG5CnoSWLfz0oKwVV+AnHmYSQi2aAY1rI5HQHimSR9f/Ww8liN+2pegaVdk5hu7X3B7WztP5+LDsZ8zcqeiawVT1WvE4p+A=="
    }
}
```

Here are the monthly stats on how many times I've received this particular message:

```csv
YEAR	MONTH	MESSAGES
2022	11	272
2022	10	805
2022	9	780
2022	8	806
2022	7	807
2022	6	779
2022	5	996
2022	4	603
2022	3	810
2022	2	982
2022	1	807
2021	12	780
2021	11	782
2021	10	787
2021	9	694
2021	8	281
2021	7	536
2021	6	301
2021	5	249
2021	4	26
```

Other instances don't seem to be doing this, at least with the bits of spot checking I did of my logs. I checked another account on `mastodon.cc` and it has a similar problem.

```csv
YEAR	MONTH	MESSAGES
2022	11	269
2022	10	806
2022	9	780
2022	8	806
2022	7	803
2022	6	779
2022	5	997
2022	4	604
2022	3	811
2022	2	579
2022	1	26
```

I don't know if this is a bug in Mastodon, or somehow a misconfiguration of this particular instance. Any help is appreciated.

### Specifications

Unknown, has been happening since at least April 2021.

**Comment by ClearlyClaire** [comment]
That is a weird one, and it doesn't raise a bell… my understanding is that something goes wrong when trying to delete this user, so the deletion procedure restarts from scratch every time.

**Comment by aaronpk** [comment]
Is there retry logic on activitypub delivery? If so, what indicates a failure? I'm pretty sure I'm responding with a success code the same as every other instance. At the very least I can confirm I'm not rejecting it with an invalid signature error.

**Comment by ClearlyClaire** [comment]
No, what happens is the delete job queues the ActivityPub delivery jobs, so a failure in the deletion job will make the whole job restart and re-queue ActivityPub delivery jobs

**Comment by aaronpk** [comment]
What constitutes a failure in the deletion job? Would something that my server returns cause mastodon.cc to think the job failed?

**Comment by ClearlyClaire** [comment]
A failure in the deletion job would be a database error of some kind, a bug in Mastodon… but really something internal to `mastodon.cc`.

An error in the delivery job itself would be caused by your server returning something else than a 2xx HTTP status code, but I am only mentioning that for completeness because I am fairly sure the issue is in the deletion job (and thus not caused in any way by your server) and not the delivery job (which would result to 8 retries at most).
